### PR TITLE
Testnet reset

### PR DESCRIPTION
**Pull request by jcvernaleo**
This adds a new testnet2 network with new genesis block, premine address, magic number, etc.

Corrected some testnet related comments while there.

### CODE REVIEW

**Comment by jrick** [comment]
I thought the plan was to create a new "testnet2" (a la the testnet3 currently used by bitcoin) by just adding a new network and switching everything to use the new parameters?

**Comment by jcvernaleo** [comment]
@jrick good point.  That is why I did the incomplete PR sooner rather than later so we can catch what I did wrong.  We can discuss in chat.

**Comment by davecgh** [comment]
Since we want to the keep the result of the vote on the old testnet, the following diff is also needed:

```
diff --git a/blockchain/chain.go b/blockchain/chain.go
index bb36f7b..7077e3e 100644
--- a/blockchain/chain.go
+++ b/blockchain/chain.go
@@ -2193,8 +2193,9 @@ func (b *BlockChain) BestSnapshot() *BestState {
 //
 // This function MUST be called with the chain state lock held (for reads).
 func (b *BlockChain) maxBlockSize(prevNode *blockNode) (int64, error) {
-       // Hard fork voting on block size is not enabled on mainnet.
-       if b.chainParams.Net == wire.MainNet {
+       // Hard fork voting on block size is only enabled on testnet v1 and
+       // simnet.
+       if b.chainParams.Net != wire.TestNet && b.chainParams.Net != wire.SimNet {
                return int64(b.chainParams.MaximumBlockSizes[0]), nil
        }

diff --git a/chaincfg/params.go b/chaincfg/params.go
index 5f1cc42..9a0f3e4 100644
--- a/chaincfg/params.go
+++ b/chaincfg/params.go
@@ -730,7 +730,7 @@ var TestNet2Params = Params{
        ReduceMinDifficulty:      false,
        MinDiffReductionTime:     0, // Does not apply since ReduceMinDifficulty false
        GenerateSupported:        true,
-       MaximumBlockSizes:        []int{1000000, 1310720},
+       MaximumBlockSizes:        []int{1310720},
        MaxTxSize:                1000000,
        TargetTimePerBlock:       time.Minute * 2,
        WorkDiffAlpha:            1,
@@ -759,36 +759,6 @@ var TestNet2Params = Params{
        RuleChangeActivationMultiplier: 3,    // 75%
        RuleChangeActivationDivisor:    4,
        RuleChangeActivationInterval:   5040, // 1 week
-       Deployments: map[uint32][]ConsensusDeployment{
-               4: {{
-                       Vote: Vote{
-                               Id:          VoteIDMaxBlockSize,
-                               Description: "Change maximum allowed block size from 1MB to 1.25MiB",
-                               Mask:        0x0006, // Bits 1 and 2
-                               Choices: []Choice{{
-                                       Id:          "abstain",
-                                       Description: "abstain voting for change",
-                                       Bits:        0x0000,
-                                       IsIgnore:    true,
-                                       IsNo:        false,
-                               }, {
-                                       Id:          "no",
-                                       Description: "reject changing max allowed block size",
-                                       Bits:        0x0002, // Bit 1
-                                       IsIgnore:    false,
-                                       IsNo:        true,
-                               }, {
-                                       Id:          "yes",
-                                       Description: "accept changing max allowed block size",
-                                       Bits:        0x0004, // Bit 2
-                                       IsIgnore:    false,
-                                       IsNo:        false,
-                               }},
-                       },
-                       StartTime:  1486598400, // Feb 9th, 2017
-                       ExpireTime: 1496966400, // Jun 9th, 2017
-               }},
-       },

        // Enforce current block version once majority of the network has
        // upgraded.
```

File: chaincfg/genesis.go [hunk 1]
```diff
@@ -159,6 +159,77 @@ var testNetGenesisBlock = wire.MsgBlock{
...
+
+// testNetGenesisMerkleRoot is the hash of the first transaction in the genesis block
+// for the test network.
+var testNet2GenesisMerkleRoot = genesisCoinbaseTx2Legacy.TxHash()
+
+// testNetGenesisBlock defines the genesis block of the block chain which
+// serves as the public transaction ledger for the test network (version 3).
+var testNet2GenesisBlock = wire.MsgBlock{
+	Header: wire.BlockHeader{
+		Version:      1,
```
Comment (davecgh) [inline chaincfg/genesis.go]: Let's make this version 4 since all blocks on the new testnet are v4.

File: params.go [hunk 2]
```diff
@@ -40,6 +40,14 @@ var testNetParams = params{
 	rpcPort: "19109",
 }
 
+// testNetParams contains parameters specific to the test network (version 2)
+// (wire.TestNet2).  NOTE: The RPC port is intentionally different than the
```
Comment (davecgh) [inline params.go]: We can remove this note since dcrd is the reference implementation.

File: chaincfg/genesis.go [hunk 3]
```diff
@@ -159,6 +159,77 @@ var testNetGenesisBlock = wire.MsgBlock{
 // test network.
 var testNetGenesisHash = testNetGenesisBlock.BlockHash()
 
+// TestNet2 ------------------------------------------------------------------------
+
+// genesisCoinbaseTx2Legacy is the coinbase transaction for the genesis block for
+// the test network.
+var genesisCoinbaseTx2Legacy = wire.MsgTx{
```
Comment (davecgh) [inline chaincfg/genesis.go]: I would suggest just reusing the existing `genesisCoinbaseTxLegacy` used by the other testnet network.  These are identical and changing the date in the new genesis block header is sufficient to make it unique.

File: chaincfg/genesis.go [hunk 4]
```diff
@@ -159,6 +159,77 @@ var testNetGenesisBlock = wire.MsgBlock{
...
+			},
+		},
+	},
+	LockTime: 0,
+	Expiry:   0,
+}
+
+// testNetGenesisMerkleRoot is the hash of the first transaction in the genesis block
+// for the test network.
+var testNet2GenesisMerkleRoot = genesisCoinbaseTx2Legacy.TxHash()
```
Comment (davecgh) [inline chaincfg/genesis.go]: Same here.

File: chaincfg/genesis.go [hunk 5]
```diff
@@ -159,6 +159,77 @@ var testNetGenesisBlock = wire.MsgBlock{
...
+
+// testNetGenesisBlock defines the genesis block of the block chain which
+// serves as the public transaction ledger for the test network (version 3).
+var testNet2GenesisBlock = wire.MsgBlock{
+	Header: wire.BlockHeader{
+		Version:      1,
+		PrevBlock:    chainhash.Hash{},
+		MerkleRoot:   testNetGenesisMerkleRoot,
```
Comment (davecgh) [inline chaincfg/genesis.go]: This really should have been `testNet2GenesisMerkleRoot` with the current changes, however, I would suggest leaving this as `testNetGenesisMerkleRoot` and just removing the others as mentioned above.

File: chaincfg/genesis.go [hunk 6]
```diff
@@ -159,6 +159,77 @@ var testNetGenesisBlock = wire.MsgBlock{
...
+		Timestamp:    time.Unix(1489550400, 0), // 2017-03-15 TestNet10
+		Bits:         0x1e00ffff,
+		SBits:        20000000,
+		Nonce:        0x18aea41a,
+		StakeVersion: 0,
+	},
+	Transactions: []*wire.MsgTx{&genesisCoinbaseTx2Legacy},
```
Comment (davecgh) [inline chaincfg/genesis.go]: Then this would be `&genesisCoinbaseTxLegacy`.

File: chaincfg/genesis.go (hunk 3 above)
Comment (jcvernaleo) [inline chaincfg/genesis.go]: done

File: chaincfg/genesis.go (hunk 4 above)
Comment (jcvernaleo) [inline chaincfg/genesis.go]: done

File: chaincfg/genesis.go (hunk 6 above)
Comment (jcvernaleo) [inline chaincfg/genesis.go]: done

File: chaincfg/genesis.go (hunk 5 above)
Comment (jcvernaleo) [inline chaincfg/genesis.go]: left along since i did the change above

File: chaincfg/genesis.go (hunk 1 above)
Comment (jcvernaleo) [inline chaincfg/genesis.go]: done

File: params.go (hunk 2 above)
Comment (jcvernaleo) [inline params.go]: done

**Pull request merged**